### ISSUE TITLE
Creating/enabling currency

### ISSUE THREAD

**Issue by pareeohnos** [waynerobinson/xeroizer#401]
Is it not possible to enable a currency in an organisation? The currency model has `read_only` permission, and I get an error (completely unrelated to that) if I attempt to build a currency record and save it, but I can't see any other way of trying to enable a currency.

Is this an oversight, or a limitation of Xeroizer? Alternatively if I am mis-interpreting the code, is there a bug in the creation? As the error I receive is not one telling me there is no permission to save, but something completely unrelated

```ruby
TypeError - nil is not a symbol:
   () Users/Aidy/.rvm/gems/ruby-2.1.3/bundler/gems/xeroizer-7efa2088818f/lib/xeroizer/record/base.rb:54:in `[]'
   () Users/Aidy/.rvm/gems/ruby-2.1.3/bundler/gems/xeroizer-7efa2088818f/lib/xeroizer/record/model_definition_helper.rb:88:in `id'
   () Users/Aidy/.rvm/gems/ruby-2.1.3/bundler/gems/xeroizer-7efa2088818f/lib/xeroizer/record/base.rb:81:in `new_record?'
   () Users/Aidy/.rvm/gems/ruby-2.1.3/bundler/gems/xeroizer-7efa2088818f/lib/xeroizer/record/base.rb:109:in `save'
```

This is after trying to run

    currency = @client.Currency.build
    currency.code = 'USD'
    currency.save

**Comment by rjaus** [comment]
Hey @pareeohnos 

Currencies can be 'created' only.  What actually occurs is the currency is "added" or "subscribed to" by the org.  This is done via a PUT method: https://developer.xero.com/documentation/api/currencies

It's failing because the Currency model doesn't have an ID, which leads to .new_record? raising an error.

I've created a branch that enables this behaviour by overrides the new_record? method. 

Let me know what you think: https://github.com/waynerobinson/xeroizer/blob/Currency/lib/xeroizer/models/currency.rb

**Comment by pareeohnos** [comment]
that should do the trick! I knew it was possible within Xero itself, didn't realise it was such a simple fix within Xeroizer though :)

I assume this doesn't have any further implications? Obviously just returning `true` for new_record? means it will always attempt to 'create' the currency which I'm guessing will be rejected by Xero?

**Comment by rjaus** [comment]
API will reject upon update, raising a validation error.

Could be improved, but better than lacking functionality altogether.

**Comment by pareeohnos** [comment]
Yeah agreed - I'd definitely rather have the feature and know to work around any quirks than not have it :)

**Comment by rjaus** [comment]
Resolved: #411
